testdata/e2e/e2e.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+set -e
+
+: ${IMAGE:?Need to set metrics-agent IMAGE variable to test}
+: ${KUBERNETES_VERSION:?Need to set KUBERNETES_VERSION to test}
+
+export WORKINGDIR=${TEMP_DIR}/testdata/e2e/e2e-${KUBERNETES_VERSION}
+export CI_KUBECTL="docker exec -i e2e-${KUBERNETES_VERSION}-control-plane kubectl --server=https://127.0.0.1:6443"
+
+cleanup() {
+  kind delete cluster --name=e2e-${KUBERNETES_VERSION} &> /dev/null || true
+  if [ -d $WORKINGDIR ]; then
+    echo "Cleaning up Temp directory : ${WORKINGDIR}"
+    rm -rf $WORKINGDIR
+  fi
+}
+
+setup_kind() {
+
+  export PATH=$(go env GOPATH)/bin:$PATH
+
+  cleanup
+
+  if ! (kind create cluster --name=e2e-${KUBERNETES_VERSION} --image=kindest/node:${KUBERNETES_VERSION}) ; then
+    echo "Could not create KinD cluster"
+    exit 1
+  fi
+
+  sleep 2
+  
+    i=0
+    until [ $i -ge 5 ]
+    do
+      kind load docker-image ${IMAGE} --name e2e-${KUBERNETES_VERSION} && echo "${IMAGE} image added to cluster" && break
+      n=$[$i+1]
+      sleep 15
+    done
+}
+
+deploy(){
+  mkdir -p -m 0777 ${WORKINGDIR}
+
+  if [ ! -d $WORKINGDIR ]; then
+    >&2 echo "Failed to create temp directory ${WORKINGDIR}"
+    exit 1
+  fi
+
+  export CONTAINER="\"name\": \"metrics-agent\", \"image\": \"${IMAGE}\",\"imagePullPolicy\": \"Never\""
+  export ENVS="\"env\": [{\"name\": \"CLOUDABILITY_CLUSTER_NAME\", \"value\": \"e2e\"}, {\"name\": \"CLOUDABILITY_POLL_INTERVAL\", \"value\": \"20\"} ]"
+  
+  if "${CI}" = "true"; then
```

Comment:
I was seeing an error along the lines of "command not found" on this line and others like it. If I change it to this:-

```
-  if "${CI}" = "true"; then
+  if [ "${CI}" = "true" ]; then
```

the error goes away.

Reply:
thanks . fixed